app/src/main/java/com/jerboa/feat/SwipeToAction.kt

```diff
@@ -0,0 +1,103 @@
+package com.jerboa.feat
+
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Bookmark
+import androidx.compose.material.icons.outlined.Comment
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.vectorResource
+import com.jerboa.R
+
+enum class SwipeToActionType {
+    Upvote,
+    Downvote,
+    Reply,
+    Save,
+    ;
+
+    companion object {
+        const val START_THRESHOLD = 0.05f
+
+        fun getActionToRangeList(actions: List<SwipeToActionType>): List<Pair<OpenEndRange<Float>, SwipeToActionType>> {
+            val start = START_THRESHOLD + 0.05f
+            val delta = if (actions.size > 2) 0.14f else 0.2f
+            return actions.mapIndexed { index, it ->
+                (start + delta * index)
+                    .rangeUntil(if (index == actions.size - 1) 1f else (start + delta * (index + 1))) to it
+            }
+        }
+
+        fun getDefaultLeftActions(): List<SwipeToActionType> {
+            return listOf(Reply, Save)
+        }
+
+        fun getDefaultRightActions(): List<SwipeToActionType> {
+            return listOf(Upvote, Downvote)
+        }
+    }
+
+    @Composable
+    fun getImageVector(): ImageVector {
+        return when (this) {
+            Upvote -> ImageVector.vectorResource(id = R.drawable.up_outline)
+            Downvote -> ImageVector.vectorResource(id = R.drawable.down_outline)
+            Reply -> Icons.Outlined.Comment
```

Comment:
'Comment: ImageVector' is deprecated. Use the AutoMirrored version at Icons.AutoMirrored.Outlined.Comment

Reply:
Fixed